common/src/main/java/com/wynntils/wc/objects/item/render/HighlightedItem.java

```diff
@@ -0,0 +1,11 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.wc.objects.item.render;
+
+import com.wynntils.mc.event.SlotRenderEvent;
+
+public interface HighlightedItem {
+    int getHighlightColor(SlotRenderEvent e);
```

Comment:
I don't get why these events exist in all interfaces as arguments, while they are not registered? Even if they would be registered, what's the use? As far as I can tell, they aren't used in the overrides

Reply:
Oh, I think I get it, they are to be called from event methods? I think they are somewhat confusing though on the first look though.

Reply:
yeah, the purpose is to pass the event information into the method, in case some of it is necessary to determine the color/text/etc. i didn't actually use it anywhere yet, but i was planning on implementing the duplicate cosmetic filter this way - maybe there's a better method

Reply:
Yeah, on second look it is much easier to read. How do you feel about passing the `Screen` and `Slot` instead of `SlotRenderEvent` (and same with the hotbar event one). This wouldn't change anything fundamental, though the methods could be called from more places and also would not suggest that they are meant to be events?

Reply:
that's probably good for clarity's sake, i'll do that.

and honestly on further thought, the argument is not really necessary for the hotbar highlights - i can't think of any logic that would depend on the coordinate values of the hotbar slot, and the only other information is the itemstack, which is obviously already known